File: android/src/main/kotlin/com/sabiantech/sabian_native_common/channels/methods/ExampleChannelHandler.kt
package com.sabiantech.sabian_native_common.channels.methods

import com.sabiantech.sabian_native_common.channels.methods.example.ExampleChannelMethodHandler

class ExampleChannelHandler : AbstractMethodChannelHandler() {
    override fun register() {
        register("getPlatformVersion", ExampleChannelMethodHandler())
    }
}

File: ios/Classes/channels/methods/ExampleChannelHandler.swift
//
//  ExampleChannelHandler.swift
//  sabian_native_common
//
//  Created by Sweet Pea on 30/11/2023.
//

import Foundation

class ExampleChannelHandler : AbstractMethodChannelHandler {
    override func register() {
        register("getPlatformVersion", ExampleMethodChannelHandler())
    }
}

class ExampleMethodChannelHandler : IMethodChannelHandler {
    func execute(payload: MethodChannelPayload) {
        payload.result("iOS " + UIDevice.current.systemVersion)
    }
}
